### PR TITLE
feat(transfer): update transfer flow

### LINKED ISSUE #2081: Transfer (v50)
This new feature is part of the v50 [#2084 ](https://github.com/synonymdev/bitkit/issues/2084)

New user: https://www.figma.com/proto/ltqvnKiejWj0JQiqtDf2JJ/Bitkit-Wallet?page-id=21088%3A222905&node-id=21088-228002&viewport=18%2C229%2C0.13&t=CrpfDpnzVxZoIYii-8&scaling=min-zoom&content-scaling=fixed&starting-point-node-id=21088%3A228002&show-proto-sidebar=1&hide-ui=1
Active user: https://www.figma.com/proto/ltqvnKiejWj0JQiqtDf2JJ/Bitkit-Wallet?page-id=21088%3A222905&node-id=21088-228007&viewport=18%2C229%2C0.13&t=CrpfDpnzVxZoIYii-8&scaling=min-zoom&content-scaling=fixed&starting-point-node-id=21088%3A228007&show-proto-sidebar=1&hide-ui=1

**Transfer:** The Lightning Slider UI is being retired. All Transfer features and flows have been redesigned and optimized (details discussed in UX call).

### LINKED ISSUE #2159: [Bug]: Max channel duration 52 weeks (instead of 12)
### Describe the bug

Max duration for a LN channel on Bitkit is 52 weeks, instead of 12 weeks (as stated in ToS)

### Reproduce

Go to Settings > Advanced > Lightning Connections > try setting up a normal channel, select amount and size of channel then adjust channel duration. According to the ToS the max term should be 12 weeks. On v 132 I am able to set it up to 52 weeks (the maximum allowed on Bitkit in this current version) 

### Screenshots / Recording

https://github.com/user-attachments/assets/3699d68c-1f2b-4417-9c3b-32307e724072

### Operating system

Graphene OS Pixel 6A

### Bitkit version

v132

### Log output

_No response_

### PR DESCRIPTION
**Pull request by pwltr**
### Description

Update transfer UI from design updates v50

### Linked Issues/Tasks

Closes https://github.com/synonymdev/bitkit/issues/2081 https://github.com/synonymdev/bitkit/issues/2159

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (improving code without creating new functionality)

### Tests

- [x] Detox test
- [ ] Unit test
- [ ] No test

### Screenshot / Video

https://github.com/user-attachments/assets/2301bafd-6c65-49ba-8d08-82d4880f1042

https://github.com/user-attachments/assets/50b34f46-fc25-4eed-bff8-d11302be5552

### CODE REVIEW

File: src/screens/Transfer/Availability.tsx [hunk 1]
```diff
@@ -7,60 +7,39 @@ import { View as ThemedView } from '../../styles/components';
...
 
 const Availability = ({
 	navigation,
 }: TransferScreenProps<'Availability'>): ReactElement => {
 	const { t } = useTranslation('lightning');
-	const [isLoading, setIsLoading] = useState(false);
-	const dispatch = useAppDispatch();
 
 	const onCancel = (): void => {
-		navigation.goBack();
```
Comment (limpbrains) [inline src/screens/Transfer/Availability.tsx:27]: If I understand correctly, if case of goBack you remove screen from the history, when you do .navigate('Wallet') you add new screen old one remains, maybe it is better to use goBack, where possible, to reduce memory usage?

File: src/screens/Transfer/SavingsProgress.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+				await sleep(5000); // give beignet some time to process
+				navigation.navigate('Success', { type: 'savings' });
+			} else {
+				dispatch(startCoopCloseTimer());
+				navigation.navigate('Interrupted');
+			}
+		};
+
+		close();
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (limpbrains) [inline src/screens/Transfer/SavingsProgress.tsx:41]: can you please add a comment why something is not added into dependecies array?

File: src/screens/Transfer/SavingsProgress.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+	}, []);
+
+	const onClose = (): void => {
+		navigation.navigate('Wallet');
```
Comment (limpbrains) [inline src/screens/Transfer/SavingsProgress.tsx:45]: can we use goBack here?

File: src/screens/Transfer/SpendingAdvanced.tsx [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+			}
+
+			setFeeEstimate((value) => ({
+				...value,
+				[`${clientBalance}-${lspBalance}`]: result.value,
+			}));
+		};
+
+		getChannelOpenCost();
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (limpbrains) [inline src/screens/Transfer/SpendingAdvanced.tsx:96]: same

File: src/screens/Transfer/SpendingAdvanced.tsx [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+			zeroConfPayment: clientBalance <= max0ConfClientBalanceSat,
+		});
+
+		setLoading(false);
+
+		if (response.isErr()) {
+			const { message } = response.error;
+			// const description = message.includes('Local channel balance is too small')
+			// 	? t('error_channel_receiving', { usdValue: maxLspBalance })
+			// 	: t('error_channel_setup_msg', { raw: message });
```
Comment (limpbrains) [inline src/screens/Transfer/SpendingAdvanced.tsx]: can this be removed?

File: src/screens/Transfer/SpendingAdvanced.tsx [hunk 6]
```diff
@@ -0,0 +1,324 @@
...
+						onPress={onChangeUnit}
+					/>
+				</View>
+
+				<View style={styles.fee}>
+					<Caption13Up style={styles.feeLabel} color="secondary">
+						{t('spending_advanced.fee')}:
+					</Caption13Up>
+					<Money
+						sats={feeEstimate[`${clientBalance}-${lspBalance}`] || 0}
```
Comment (limpbrains) [inline src/screens/Transfer/SpendingAdvanced.tsx]: I think it should not be 0 by default. Better to show loading indicator, or dash, IDK

**Review by limpbrains — CHANGES_REQUESTED** [review]

File: src/screens/Transfer/Availability.tsx (hunk 1 above)
Comment (pwltr) [reply src/screens/Transfer/Availability.tsx:27]: From what I know if a screen is already in the stack `navigate` will remove all the screens after it. `goBack` & `popToTop` didn't work in this case.

See: https://reactnavigation.org/docs/navigation-prop/#navigate

File: src/screens/Transfer/SavingsProgress.tsx (hunk 2 above)
Comment (pwltr) [reply src/screens/Transfer/SavingsProgress.tsx:41]: there is a comment above the `useEffect` "on mount". Should be clear? Or should we move the comment down?

File: src/screens/Transfer/SpendingAdvanced.tsx (hunk 5 above)
Comment (pwltr) [reply src/screens/Transfer/SpendingAdvanced.tsx]: removed

File: src/screens/Transfer/SpendingAdvanced.tsx (hunk 6 above)
Comment (pwltr) [reply src/screens/Transfer/SpendingAdvanced.tsx]: done

File: src/utils/lightning/index.ts [hunk 7]
```diff
@@ -1298,31 +1298,29 @@ export const closeChannel = async ({
...
+	channels: TChannel[];
 	force?: boolean; // It will always try to coop close first and only force close if set to true.
-} = {}): Promise<Result<TChannel[]>> => {
-	const selectedWallet = getSelectedWallet();
-	const selectedNetwork = getSelectedNetwork();
-
+}): Promise<Result<TChannel[]>> => {
 	try {
 		// Ensure we're fully up-to-date.
-		const refreshRes = await refreshLdk({ selectedWallet, selectedNetwork });
```
Comment (limpbrains) [inline src/utils/lightning/index.ts:1317]: ❤️

**Review by limpbrains — APPROVED** [review]

**Pull request merged**